bin/rbenv-sh-update

```diff
@@ -4,33 +4,37 @@ set -e
 
 shell="$(basename "${RBENV_SHELL:-$SHELL}")"
 
-case "$shell" in
-fish )
-  cat <<EOF
-command rbenv update $@
-if command test -t 1
-  printf "\\e[1;32mreloading rbenv\\e[0m\\n"
-  . (rbenv init -|psub)
-  printf " \\033[1;32m|\\033[0m  done\\n"
-else
-  printf "reloading rbenv\\n"
-  . (rbenv init -|psub)
-  printf " |  done\\n"
-end
-EOF
-  ;;
-* )
-  cat <<EOF
-command rbenv update $@;
-if [ -t 1 ]; then
-  printf "\\e[1;32mreloading rbenv\\e[0m\\n";
-  eval "\$(rbenv init -)";
-  printf " \\033[1;32m|\\033[0m  done\\n";
-else
-  printf "reloading rbenv\\n";
-  eval "\$(rbenv init -)";
-  printf " |  done\\n";
-fi
-EOF
-  ;;
+# Parse command line flags
+case "$1" in
+  -n | --noop ) noop=true ;;
+  -q | --quiet | --silent ) quiet=true ;;
 esac
+
+
+
+echo "if command rbenv update $@; then"
+
+if [ -z "$quiet" ]; then
+  echo "  if command test -t 1; then"
+  echo "    printf \"\\e[1;32mReloading rbenv\\e[0m\\n\";"
+  echo "  else"
+  echo "    printf \"Reloading rbenv\\n\";"
+  echo "  fi;"
+fi
+
+if [ -z "$noop" ]; then
+  case "$shell" in
+  fish ) echo "  . (rbenv init -|psub)" ;;
```

Comment:
This should be `source` since  `.` has been deprecated, see https://github.com/fish-shell/fish-shell/issues/310 and http://fishshell.com/release_notes.html